### PR TITLE
Create SELU Many Layers

### PR DESCRIPTION
**Pull request by michael-GitHub-0**
I am new to GitHub, so I may be doing this incorrectly. I wanted to share some code to compare ReLU vs ELU vs SELU on the MNIST data (Chapter 10 and 11). The code isn't interesting (it's all from the book) but I think the results are interesting. With 50 hidden layers, ReLU and ELU can learn if they use He initialization, but the performance is poor; however, SELU learns as well as a shallow network even without He initialization.

The book is amazing! The effort the author put in to help readers learn and understand is obvious, and very much appreciated! I am looking forward to future editions!

### CODE REVIEW

**Comment by michael-GitHub-0** [comment]
Also, I am wondering if the below code could be reviewed for a coding or conceptual error. Using the SELU activation function with 50 hidden layers and 0.99 dropout, I get perfect accuracy with the training, test, and validation data. I am assuming I have made an error but when I use less extreme hyperparameters, the accuracy is about as expected; so I am not sure what the error might be. I can imagine why a deep network using the SELU activation function and an extreme dropout could force a network to learn a complex, but generalizeable feature hierarchy, but perfect accuracy seems unlikely. 

import tensorflow as tf

from tensorflow.python.framework import ops
ops.reset_default_graph()

from datetime import datetime
from datetime import timezone

def utc_to_local(utc_dt):
    return utc_dt.replace(tzinfo=timezone.utc).astimezone(tz=None)

now_day = utc_to_local(datetime.utcnow()).strftime("%Y%m%d")
now_time = utc_to_local(datetime.utcnow()).strftime("%H%M%S")
root_logdir = "/2017 07 - TensorFlow"
logdir = "{}/TensorBoard/Run-{}-{}/".format(root_logdir, now_day, now_time)

n_inputs = 28*28
n_outputs = 10

learning_rate = 0.01

X = tf.placeholder(tf.float32, shape=(None, n_inputs), name="X")
y = tf.placeholder(tf.int64, shape=(None), name="y")

X_valid = mnist.validation.images
y_valid = mnist.validation.labels

training = tf.placeholder_with_default(False, shape=(), name='training')

def selu(z, scale=1.0507009873554804934193349852946, alpha=1.6732632423543772848170429916717):
    return scale * tf.where(z >= 0.0, z, alpha * tf.nn.elu(z))

def dnn(inputs, n_hidden_layers=50, n_neurons=50, dropout_rate=0.99, name=None, activation=selu):
    with tf.variable_scope(name, "dnn"):
        for layer in range(n_hidden_layers):
            if dropout_rate:
                inputs = tf.layers.dropout(inputs, dropout_rate, training=training)
            inputs = tf.layers.dense(inputs, n_neurons, activation=activation, name="hidden%d" % (layer + 1))
        return inputs

with tf.name_scope("dnn"):
    dnn_outputs = dnn(X)
    logits = tf.layers.dense(dnn_outputs, n_outputs, activation=selu, name="logits")
    
with tf.name_scope("loss"):
    xentropy = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=y, logits=logits)
    loss = tf.reduce_mean(xentropy, name="loss")
    loss_summary = tf.summary.scalar('log_loss', loss)
    
with tf.name_scope("train"):
    optimizer = tf.train.GradientDescentOptimizer(learning_rate)
    training_op = optimizer.minimize(loss)
    
with tf.name_scope("eval"):
    correct = tf.nn.in_top_k(logits, y, 1)
    accuracy = tf.reduce_mean(tf.cast(correct, tf.float32))
    accuracy_summary = tf.summary.scalar('accuracy', accuracy)
    
init = tf.global_variables_initializer()
file_writer = tf.summary.FileWriter(logdir, tf.get_default_graph())
saver = tf.train.Saver()

n_epochs = 5
batch_size = 50

with tf.Session() as sess:
    
    init.run()
    
    for epoch in range(n_epochs):
        for iteration in range(mnist.train.num_examples // batch_size):
            X_batch, y_batch = mnist.train.next_batch(batch_size)
            sess.run(training_op, feed_dict={training: True, X: X_batch, y: y_batch})
        acc_train = accuracy.eval(feed_dict={X: X_batch, y: y_batch})
        acc_test = accuracy.eval(feed_dict={X: mnist.test.images, y: mnist.test.labels})
        
        acc_val, loss_val, accuracy_summary_str, loss_summary_str = sess.run(
            [accuracy, loss, accuracy_summary, loss_summary], feed_dict={X: X_valid, y: y_valid})
        
        file_writer.add_summary(accuracy_summary_str, epoch)
        file_writer.add_summary(loss_summary_str, epoch)
        
        print(epoch + 1, "Train accuracy:", acc_train, "Test accuracy:", acc_test, "Valid accuracy:", acc_val)
        
    save_path = saver.save(sess, "{}/Savers/Run-{}-{}/Saver.ckpt".format(root_logdir, now_day, now_time))

file_writer.close()
